### ISSUE TITLE
esxi provider always wants to change virtual_disk_id even if it is not necessary

### ISSUE THREAD

**Issue by mj84** [josenk/terraform-provider-esxi#48]
Hello there,

first of thanks alot for this awesome provider, really makes my life alot easier :)
I'm using an atomic disk layout for my linux-based VMs:
1. HDD: 2 Partitions for /boot and /
2. HDD: 1 Partition for swap
3. HDD: 1 Partition for /home
4. HDD: 1 Partition for /tmp
5. HDD: 1 Partition for /var
6. HDD: 1 Partition for /var/log

For some reason, the esxi provider always reports the following change when running `terraform plan`:
```
  ~ esxi_guest.my-guest
      virtual_disks.#:                 "3" => "5"
      virtual_disks.3.slot:            "" => "0:4"
      virtual_disks.3.virtual_disk_id: "" => "/vmfs/volumes/DS001/Terraform/my-guest_var.vmdk"
      virtual_disks.4.slot:            "" => "0:5"
      virtual_disks.4.virtual_disk_id: "" => "/vmfs/volumes/DS001/Terraform/my-guest_log.vmdk"
```

As far as I can tell, there is no change happening when running `terraform apply`

Cheers

**Comment by josenk** [comment]
How is this deployed?  Imported, Cloned VM, or you specified a source VMX.OVF directory?
I'm not sure it will support having multiple virtual disks for the OS???   I never tried that.

You would have to send me the complete .tf files.

  You may want to compare "terraform show" before and after the "terraform apply" to see if anything changed.
   Does the my-guest.vmx file on the esxi host change after the terraform apply?

**Comment by mj84** [comment]
I am creating the bare-metal VMs here, as I will be installing them via PXE-boot and kickstart.
At first, I tried cloning a bare-metal VM which has my desired disk layout, but I noticed that only the first HDD got cloned, but that might be a different issue.

I have used the following .tf for my VM, I just removed the provider settings:
```
resource "esxi_virtual_disk" "my-guest_swap" {
  virtual_disk_disk_store = "DS001"
  virtual_disk_dir        = "Terraform"
  virtual_disk_name       = "my-guest_swap.vmdk"
  virtual_disk_size       = 1
  virtual_disk_type       = "zeroedthick"
}
resource "esxi_virtual_disk" "my-guest_home" {
  virtual_disk_disk_store = "DS001"
  virtual_disk_dir        = "Terraform"
  virtual_disk_name       = "my-guest_home.vmdk"
  virtual_disk_size       = 1
  virtual_disk_type       = "zeroedthick"
}
resource "esxi_virtual_disk" "my-guest_tmp" {
  virtual_disk_disk_store = "DS001"
  virtual_disk_dir        = "Terraform"
  virtual_disk_name       = "my-guest_tmp.vmdk"
  virtual_disk_size       = 1
  virtual_disk_type       = "zeroedthick"
}
resource "esxi_virtual_disk" "my-guest_var" {
  virtual_disk_disk_store = "DS001"
  virtual_disk_dir        = "Terraform"
  virtual_disk_name       = "my-guest_var.vmdk"
  virtual_disk_size       = 2
  virtual_disk_type       = "zeroedthick"
}
resource "esxi_virtual_disk" "my-guest_log" {
  virtual_disk_disk_store = "DS001"
  virtual_disk_dir        = "Terraform"
  virtual_disk_name       = "my-guest_log.vmdk"
  virtual_disk_size       = 2
  virtual_disk_type       = "zeroedthick"
}
resource "esxi_guest" "my-guest" {
  guest_name         = "my-guest"
  disk_store         = "DS001"
  power              = "off"
  guestos            = "centos7-64"
  numvcpus           = "2"
  memsize            = "512"
  virthwver          = "14"
  boot_disk_type     = "zeroedthick"
  boot_disk_size     = 8
  network_interfaces = [
    {
      virtual_network = "Net1"
    },
    {
      virtual_network = "Net2"
    },
    {
      virtual_network = "Net3"
    },
    {
      virtual_network = "Net4"
    },
  ]
  virtual_disks = [
    {
      virtual_disk_id = "${esxi_virtual_disk.my-guest_swap.id}"
      slot            = "0:1"
    },
    {
      virtual_disk_id = "${esxi_virtual_disk.my-guest_home.id}"
      slot            = "0:2"
    },
    {
      virtual_disk_id = "${esxi_virtual_disk.my-guest_tmp.id}"
      slot            = "0:3"
    },
    {
      virtual_disk_id = "${esxi_virtual_disk.my-guest_var.id}"
      slot            = "0:4"
    },
    {
      virtual_disk_id = "${esxi_virtual_disk.my-guest_log.id}"
      slot            = "0:5"
    },
  ]
}
```
When running `terraform apply` again, the .vmx does in fact change, but only regarding empty lines and some lines are being moved around.

```
[root@localhost:/vmfs/volumes/5cd04d8f-2d0e8f21-c545-bc305bf1a350/my-guest] diff my-guest.vmx my-guest.vmx.bak
--- my-guest.vmx
+++ my-guest.vmx.bak
@@ -28,45 +28,31 @@
 ide1:0.present = "TRUE"
 ide1:0.fileName = ""
 ide1:0.deviceType = "cdrom-image"
-ethernet0.networkName = "Net1"
-ethernet0.virtualDev = "e1000"
-ethernet0.present = "TRUE"
-ethernet1.networkName = "Net2"
-ethernet1.virtualDev = "e1000"
-ethernet1.present = "TRUE"
-ethernet2.networkName = "Net3"
-ethernet2.virtualDev = "e1000"
-ethernet2.present = "TRUE"
-ethernet3.networkName = "Net4"
-ethernet3.virtualDev = "e1000"
-ethernet3.present = "TRUE"
 scsi0:1.deviceType = "scsi-hardDisk"
-
-
 scsi0:1.fileName = "/vmfs/volumes/NVMe4/Terraform/my-guest_swap.vmdk"
 scsi0:1.present = "true"
-
 scsi0:2.deviceType = "scsi-hardDisk"
-
-
 scsi0:2.fileName = "/vmfs/volumes/NVMe4/Terraform/my-guest_home.vmdk"
 scsi0:2.present = "true"
-
 scsi0:3.deviceType = "scsi-hardDisk"
-
-
 scsi0:3.fileName = "/vmfs/volumes/NVMe4/Terraform/my-guest_tmp.vmdk"
 scsi0:3.present = "true"
-
 scsi0:4.deviceType = "scsi-hardDisk"
-
-
 scsi0:4.fileName = "/vmfs/volumes/NVMe4/Terraform/my-guest_var.vmdk"
 scsi0:4.present = "true"
-
 scsi0:5.deviceType = "scsi-hardDisk"
-
-
 scsi0:5.fileName = "/vmfs/volumes/NVMe4/Terraform/my-guest_log.vmdk"
 scsi0:5.present = "true"
+ethernet0.networkName = "Net1"
+ethernet0.virtualDev = "e1000"
+ethernet0.present = "TRUE"
+ethernet1.networkName = "Net2"
+ethernet1.virtualDev = "e1000"
+ethernet1.present = "TRUE"
+ethernet2.networkName = "Net3"
+ethernet2.virtualDev = "e1000"
+ethernet2.present = "TRUE"
+ethernet3.networkName = "Net4"
+ethernet3.virtualDev = "e1000"
+ethernet3.present = "TRUE"
```

**Comment by josenk** [comment]
This should be fixed in 1.4.3...
>https://github.com/josenk/terraform-provider-esxi/releases/tag/1.4.3

**Comment by mj84** [comment]
Great, I can confirm this behaviour is fixed in 1.4.3.
Thanks alot for the quick fix :)
